src/components/progress-bar/index.js

```diff
@@ -0,0 +1,90 @@
+import React, { useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { STATUSES } from '../../constants';
+
+import classes from './styles.scss';
+
+const percentFormatter = new Intl.NumberFormat('default', {
+  style: 'percent',
+  maximumFractionDigits: 2,
+});
+
+const getFontStyles = (font) => {
+  return (
+    font && {
+      fontSize: font?.fontSize,
+      color: font?.color,
+      fontWeight: font?.fontWeight,
+    }
+  );
+};
+
+const ProgressBar = ({
+  max = 0,
+  value = 0,
+  status,
+  onEnd,
+  label,
+  font,
+  backgroundColor,
+  barColor,
+  height,
+}) => {
+  const progressBar = useRef(null);
+  const fontStyles = getFontStyles(font);
+
+  useEffect(() => {
+    const progressPct =
+      max && value ? percentFormatter.format(value / max) : '0px';
+    progressBar.current.style.width = progressPct;
+  }, [max, value]);
+
+  const transitionHandler = () => {
+    if (value === max && onEnd) onEnd();
+  };
+
+  return (
+    <div
+      className={`${classes['progress']}`}
+      style={{ backgroundColor, height }}
+    >
+      <div
+        className={`${classes['counter']} ${classes[status]}`}
+        style={{
+          ...(onEnd && { transition: 'width 3s' }),
+          backgroundColor: barColor,
+          height,
+        }}
+        ref={progressBar}
+        onTransitionEnd={transitionHandler}
+      >
+        {label && (
+          <span
+            style={{ paddingLeft: '5px', lineHeight: height, ...fontStyles }}
+          >
+            {label}
+          </span>
+        )}
+      </div>
+    </div>
+  );
+};
+
+ProgressBar.STATUSES = { ...STATUSES };
```

Comment:
Just curious, does this not work?

```javascript
ProgressBar.STATUSES = STATUSES;
```

Reply:
It does! PR updated, and also the font props setting.